File: HangJungCho/1주차/PG_42748_k번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(int i =0;i<commands.size();i++){
        int start = commands[i][0];
        int end = commands[i][1];
        int ans = commands[i][2];
        
        vector<int> A;
        for(int j = start-1;j<=end-1;j++){
            A.push_back(array[j]);
        }
        sort(A.begin(),A.end());
        answer.push_back(A[ans-1]);
    }
    return answer;
}

File: Seoin0319/1_week/PG_42748_K번째수.py
def solution(array, commands):
    answer = []

    # commands의 [i, j, k]를 하나씩 처리
    for i, j, k in commands:
        # 파이썬 인덱스는 0부터 시작하므로 i는 -1 처리
        # j는 슬라이싱에서 끝 인덱스를 포함하지 않지만, j번째까지 포함하려면 그대로 사용하면 됨
        sliced = array[i - 1:j]

        # 잘라낸 배열 정렬
        sliced.sort()

        # k번째 값 추가
        # 파이썬은 0번째부터 시작하므로 k - 1
        answer.append(sliced[k - 1])

    return answer